packages/localization/src/dates/modifyDateBy.ts

```diff
@@ -7,32 +7,49 @@ import CalendarDate from "./CalendarDate.js";
  * @param date CalendarDate instance
  * @param amount how many days/months/years to add (can be a negative number)
  * @param unit what to modify: "day", "month" or "year"
+ * @param isPageUpOrPageDown whether the method is called when the date is being modified by the PageUp/PageDown keys/logic
  * @param minDate minimum date to enforce
  * @param maxDate maximum date to enforce
  */
-const modifyDateBy = (date: CalendarDate, amount: number, unit: string, minDate?: CalendarDate, maxDate?: CalendarDate) => {
+const modifyDateBy = (date: CalendarDate, amount: number, unit: string, isPageUpOrPageDown?:boolean, minDate?: CalendarDate, maxDate?: CalendarDate) => {
```

Comment:
This comment is more or less as my very first comment on the second method, that was named also after a specific user interaction (headerBtnClick or something like that). Now the parameter is bound to a user interaction (the pageUpDown press).

I was thinking for a more generic name and I can suggest:
- preserveDate/preserveDay - the way pageUpDown should work
or
-  startEndDate/startEndDay - similar to minDate and maxDate, to enforce some restrictions setting the first or the last day of the month when set to true.

Between the ending Date vs Day, you can follow the Date API wording for consistency.
The getDate()  returns the day of the month, The getDay()  returns the day of the week.
So, It's seems Date (preserveDate, startEndDate) is the right one and with we can refer to it it in the method param docs as "day of the month" to make it clear.